core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -27,13 +30,20 @@
         implements Renderer<String>
 {
     private static final JsonCodec<JsonRenderedNode> CODEC = JsonCodec.jsonCodec(JsonRenderedNode.class);
+    private static final JsonCodec<Map<PlanFragmentId, JsonPlanFragment>> PLAN_MAP_CODEC
```

Comment:
I thought a map would be better to keep all things that produce json plan in the same json map format. I am working on adding json support for explain analyze as well.

Reply:
I checked the example gist and it feels like it may not be very clear what does the key indicate. If it was a field with an explicit name (e.g.: `planFragmentId`) it could be easier to read.

Reply:
The key has to be unique. I can probably append the Id to a string to create a key `"planFragmentId1"` perhaps?

Reply:
You can add a `planFragmentId` field to the `JsonPlanFragment` class

Reply:
Added `planFragmentId` to `JsonPlanFragment` class.

Reply:
> I thought a map would be better to keep all things that produce json plan in the same json map format.

Sorry, i don't understand, could you please elaborate? Are you trying to align with an existing formatting? Could you please point me out to what it is?